lib/js/app/queryCreator/utils/showField.test.ts

```diff
@@ -0,0 +1,39 @@
+import { showField } from './showField';
+
+test('should show analysisType for all types of analysis', () => {
+  const fieldCount = showField('analysisType', 'count');
+  const fieldSum = showField('analysisType', 'sum');
+  const fieldAverage = showField('analysisType', 'average');
+
+  expect(fieldCount).toBeTruthy();
+  expect(fieldSum).toBeTruthy();
+  expect(fieldAverage).toBeTruthy();
+});
+
+test('should show percentile field only for percentile type of analysis', () => {
+  const fieldCount = showField('percentile', 'count');
+  const fieldSum = showField('percentile', 'sum');
+  const fieldAverage = showField('percentile', 'percentile');
+
+  expect(fieldCount).toBeFalsy();
+  expect(fieldSum).toBeFalsy();
+  expect(fieldAverage).toBeTruthy();
+});
+
+test('should show only supported fields for extraction', () => {
+  const analysisType = showField('analysisType', 'extraction');
+  const eventCollection = showField('eventCollection', 'extraction');
+  const targetProperty = showField('targetProperty', 'extraction');
+  const email = showField('email', 'extraction');
+  const contentEncoding = showField('contentEncoding', 'extraction');
+  const propertyNames = showField('propertyNames', 'extraction');
+  const latest = showField('latest', 'extraction');
+
+  expect(analysisType).toBeTruthy();
```

Comment:
For better - readability I suggest to refactor it to: `test.each` - check Jest [docs](https://jestjs.io/docs/en/next/api.html#testeachtablename-fn-timeout) for reference.

```
test.each([
  [1, 1, 2],
  [1, 2, 3],
  [2, 1, 3],
])('.add(%i, %i)', (a, b, expected) => {
  expect(a + b).toBe(expected);
});
```